pkg/3scale/amp/operator/redis_options_provider.go

```diff
@@ -272,3 +271,37 @@ func (r *RedisOptionsProvider) setTopologySpreadConstraints() {
 		r.options.BackendRedisTopologySpreadConstraints = r.apimanager.Spec.Backend.RedisTopologySpreadConstraints
 	}
 }
+
+func (r *RedisOptionsProvider) systemRedisPodTemplateAnnotations() map[string]string {
+	annotations := make(map[string]string)
+	for k, v := range r.apimanager.Spec.System.RedisAnnotations {
+		annotations[k] = v
+	}
+	annotations["generationID"] = r.getRedisConfigCmResourceVersion()
+	return annotations
+}
+
+func (r *RedisOptionsProvider) backendRedisPodTemplateAnnotations() map[string]string {
+	annotations := make(map[string]string)
+	for k, v := range r.apimanager.Spec.Backend.RedisAnnotations {
+		annotations[k] = v
+	}
+	annotations["generationID"] = r.getRedisConfigCmResourceVersion()
+	return annotations
+}
+
+func (r *RedisOptionsProvider) getRedisConfigCmResourceVersion() string {
+	// get resourceVersion from CM and use it as Annotation "generationID" for redis Pods
+	key := client.ObjectKey{
+		Namespace: r.namespace,
+		Name:      "redis-config",
+	}
+	cm := &v1.ConfigMap{}
+	err := r.client.Get(context.Background(), key, cm)
+	if err != nil {
+		fmt.Printf("Failed to get ConfigMap: %v\n", err)
+		return ""
```

Comment:
errors need to be propagated. Usually are temporary errors.

Reply:
Done, but not sure what means `temporary errors`. It's secutiry related, so handled as error. Hope it's ok, but if not , could you please correct me. Thank you.